### PR TITLE
Nodejs week1

### PR DESCRIPTION
**Pull request by faheem102**

### CODE REVIEW

**Review by STEP0HEN — CHANGES_REQUESTED** [review]
Great Job @faheem102. 
Very nice and thoughtful that you're making it a habit to comment your code, keep it up.
Also very good with the console.logs,  which implies you're testing your code but never push console.logs, always remember to remove them when you're done testing.

I left a few comments for things to change or improve on, let me know if something isn't clear.

Awesome job done. 💯

File: nodejs/week1/src/backend/app.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Import all the data
+const express = require("express");
+const meals = require("./data/meals");
+const reservations = require("./data/reservations");
+const reviews = require("./data/reviews");
+
+// golbal variable
+const app = express();
```
Comment (STEP0HEN) [inline nodejs/week1/src/backend/app.js:8]: did you mean to say express.Router() here?

File: nodejs/week1/src/backend/app.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  response.json(cheapMeals);
+});
+// get large meals with reviews
+app.get("/large-meals", async (request, response) => {
+  const largeMeals = allMealswithReviews.filter((meal)=>{return meal.maxNumberOfGuests > 3});
+  response.json(largeMeals);
+});
+// random meal with review
+app.get("/meal", async (request, response) => {
+  const randomMeal = allMealswithReviews[Math.floor(Math.random()* allMealswithReviews.length)];
```
[the remark this answers is not in the document]
Comment (faheem102) [reply nodejs/week1/src/backend/app.js]: I have made a separate random number generator function and used it twice to get random meal and reservation.

File: nodejs/week1/src/backend/server.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const app = require("./app");
+
+app.listen(3000, () => {
+    console.log("Example app listening on port 3000!");
+});
```
[the remark this answers is not in the document]
Comment (faheem102) [reply nodejs/week1/src/backend/server.js]: this actually came as template given by Orhan. I was not sure if I can modify it in anyway.

File: nodejs/week1/src/backend/app.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  response.json(randomMeal);
+});
+// all reservations
+app.get("/reservations", async (request, response) => {
+  const allMeals = reservations.map((res)=>{return res});
+  response.json(allMeals);
+});
```
[the remark this answers is not in the document]
Comment (faheem102) [reply nodejs/week1/src/backend/app.js]: I have removed this extra line of code and got the response from reservations

File: nodejs/week1/src/backend/app.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+// this is where you will be adding your routes
+app.get("/", async (request, response) => {
+  console.log("hi my name is faheem")
+  response.send("Result of the Application");
+});
+const allMealswithReviews = meals.map((meal)=>{
+  meal.reviews = reviews.filter((review)=>meals.id===review.mealid);
+      return meal;
+
+});
```
[the remark this answers is not in the document]
Comment (faheem102) [reply nodejs/week1/src/backend/app.js]: I have modified my code here now its showing me each meal with its relevant review. please have a look at my new commit which contains all the new changes you have suggested. thank you

**Comment by STEP0HEN** [comment]
Hi Faheem, great that you've made the changes but I think you missed adding the node_modules to your .gitignore so I can't really go past that to see the changes you made because there 1000's of files for me to scroll through to get to the actual change you made.

Do that and push again and I'll look at the changes you made.

**Comment by faheem102** [comment]
> Hi Faheem, great that you've made the changes but I think you missed adding the node_modules to your .gitignore so I can't really go past that to see the changes you made because there 1000's of files for me to scroll through to get to the actual change you made.
> 
> Do that and push again and I'll look at the changes you made.

hi Stephen i put the node_modules files then commit and push but its not showing anything in the commit. how can I commit and push the same files ????

**Comment by faheem102** [comment]
> > Hi Faheem, great that you've made the changes but I think you missed adding the node_modules to your .gitignore so I can't really go past that to see the changes you made because there 1000's of files for me to scroll through to get to the actual change you made.
> > Do that and push again and I'll look at the changes you made.
> 
> hi Stephen i put the node_modules files then commit and push but its not showing anything in the commit. how can I commit and push the same files ????

I push to the branch many times. it says everything is up to date.

File: nodejs/week1/src/backend/app.js [hunk 6]
```diff
@@ -1,3 +1,8 @@
+const app = require("./app");
+
+app.listen(3000, () => {
+    console.log("Example app listening on port 3000!");
+});
```
Comment (STEP0HEN) [inline nodejs/week1/src/backend/app.js:5]: Actually, let this be separated in the server.js. Revert this change and you can merge the PR.

**Review by STEP0HEN — APPROVED** [review]
Thanks for the node modules.
I approved the PR, revert the server js code and you can merge

**Comment by faheem102** [comment]
I have same problem Stephen and I spent more than enough time to find a solution to this I can't  sort out what to do.

**Pull request merged**